### ISSUE TITLE
Binary compatibility break in 4.1.1

### ISSUE THREAD

**Issue by aalmiray** [Twitter4J/Twitter4J#392]
Release `4.1.1` broke binary compatibility with previous releases in the 4.x series by moving some classes from `twitter4j.*` to `twitter4j.v1.*`. This release should had bumped the major version if Semantic Versioning is to be followed, making it `5.0.0`.

**Comment by aalmiray** [comment]
Apparently not only classes were moved to a different package but also renamed as well as some of their methods. `TwitterFactory` and `ConfigurationBuilder` are gone, making this previously working code now fail during compilation

https://github.com/jreleaser/jreleaser/blob/47ebff56dc2ec21e39123b99f437504d3ff6c72f/sdks/jreleaser-twitter-java-sdk/src/main/java/org/jreleaser/sdk/twitter/Twitter.java#L59-L69

**Comment by lukonjun** [comment]
I agree, I am currently trying to update this library from 4.0.7 -> 4.1.2 and it is throwing compilation errors

**Comment by 417-72KI** [comment]
+1

Also there is no release notes, migration guides, and changelogs,
It makes us very troublesome to detect where are the changes from each commits and how to apply into our codes.

**Comment by rzo1** [comment]
Without giving a try to upgrade, it is still only V1 support, right? ;-)

**Comment by bmarwell** [comment]
The recent version of Twitter4j is 4.1.2, but there was no "release", only a tag. This resulted in now different versions for twitter4j-async and twitter4j-stream, both are stuck at 4.0.7.
